server/src/test-integration/java/com/thoughtworks/go/config/FullConfigSaveFlowTestBase.java

```diff
@@ -91,17 +89,23 @@ public void shouldEncryptPluginPropertiesOfPublishTask() throws Exception {
         Configuration ancestorPluggablePublishAftifactConfigBeforeEncryption = cruiseConfig
                 .pipelineConfigByName(new CaseInsensitiveString("ancestor"))
                 .getExternalArtifactConfigs().get(0).getConfiguration();
-        assertThat(ancestorPluggablePublishAftifactConfigBeforeEncryption.getProperty("Image").getValue(), is("SECRET"));
+        assertThat(ancestorPluggablePublishAftifactConfigBeforeEncryption.getProperty("Image").getValue(), is("IMAGE_SECRET"));
         assertThat(ancestorPluggablePublishAftifactConfigBeforeEncryption.getProperty("Image").getEncryptedValue(), is(nullValue()));
-        assertThat(ancestorPluggablePublishAftifactConfigBeforeEncryption.getProperty("Image").getConfigValue(), is("SECRET"));
+        assertThat(ancestorPluggablePublishAftifactConfigBeforeEncryption.getProperty("Image").getConfigValue(), is("IMAGE_SECRET"));
 
         GoConfigHolder configHolder = getImplementer().execute(new FullConfigUpdateCommand(cruiseConfig, goConfigService.configFileMd5()), new ArrayList<>(), "Upgrade");
         Configuration ancestorPluggablePublishAftifactConfigAfterEncryption = configHolder.configForEdit
                 .pipelineConfigByName(new CaseInsensitiveString("ancestor"))
                 .getExternalArtifactConfigs().get(0).getConfiguration();
-        assertThat(ancestorPluggablePublishAftifactConfigAfterEncryption.getProperty("Image").getValue(), is("SECRET"));
+
+        assertThat(ancestorPluggablePublishAftifactConfigAfterEncryption.getProperty("Image").getValue(), is("IMAGE_SECRET"));
         assertThat(ancestorPluggablePublishAftifactConfigAfterEncryption.getProperty("Image").getEncryptedValue(), startsWith("AES:"));
         assertThat(ancestorPluggablePublishAftifactConfigAfterEncryption.getProperty("Image").getConfigValue(), is(nullValue()));
+
+        //verify xml on disk contains encrypted Image plugin property
+        assertThat(configHelper.getCurrentXml(), containsString(ancestorPluggablePublishAftifactConfigAfterEncryption.getProperty("Image").getEncryptedValue()));
+        //verify xml from GoConfigHolder contains encrypted Image plugin property
+        assertThat(getImplementer().toXmlString(configHolder.configForEdit), containsString(ancestorPluggablePublishAftifactConfigAfterEncryption.getProperty("Image").getEncryptedValue()));
```

Comment:
Add similar tests for fetch?

Reply:
@varshavaradarajan -- This particular test verifies whether the preprocessed and the configForEdit both are updated with the changeset.

And verifying one of the plugin property encryption on preprocessed and the configForEdit is good enough.

We can rely upon the entity level encryption specs and `config.encryptSecureProperties` encryption specs for the fetch plugin property encryptions.

---

Also,
If we add the test of fetch plugin properties here. Later, whenever the new GoCD entity requires plugin property encryption, we will need to add it here too.
Eg: for role-config, elastic-agent-profiles, etc. and

Reply:
The very next text is for fetch. :)